gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,351 @@
+package com.google.gcloud.storage.contrib.nio;
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.storage.BlobInfo;
+import com.google.gcloud.storage.BucketInfo;
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.testing.RemoteGcsHelper;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.ByteArrayOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Integration test for gcloud-nio. This test actually talks to GCS (you need an account).
+ * Tests both reading and writing.
+ *
+ * You *must* set the GOOGLE_APPLICATION_CREDENTIALS environment variable
+ * for this test to work. It must contain the name of a local file that contains
+ * your Service Account JSON Key.
+ *
+ * The instructions for how to get the Service Account JSON Key are
+ * at https://cloud.google.com/storage/docs/authentication?hl=en#service_accounts
+ *
+ * The short version is this: go to cloud.google.com/console,
+ * select your project, search for "API manager", click "Credentials",
+ * click "create credentials/service account key", new service account,
+ * JSON. The contents of the file that's sent to your browsers is your
+ * "Service Account JSON Key".
+ *
+ */
+@RunWith(JUnit4.class)
+public class ITGcsNio {
+
+  private static final List<String> FILE_CONTENTS = ImmutableList.of(
+      "Tous les êtres humains naissent libres et égaux en dignité et en droits.",
+      "Ils sont doués de raison et de conscience et doivent agir ",
+      "les uns envers les autres dans un esprit de fraternité.");
```

Comment:
This is from the universal declaration of human rights. The corresponding paragraph in the English version goes:

> All human beings are born free and equal in dignity and rights. They are endowed with reason and conscience and should act towards one another in a spirit of brotherhood.